### PR TITLE
[doc]Update config docstring

### PR DESCRIPTION
**Pull request by wangxiyuan**
Update the docstring for config. Add the missing args description.

### CODE REVIEW

File: vllm/config.py [hunk 1]
```diff
@@ -91,6 +91,7 @@ class ModelConfig:
             the default version.
         max_model_len: Maximum length of a sequence (including prompt and
             output). If None, will be derived from the model.
+        spec_target_max_model_len: Specific maximum length for the model.
```
Comment (DarkLight1337) [inline vllm/config.py:94]: I think this refers to speculative decoding.

File: vllm/config.py [hunk 2]
```diff
@@ -735,8 +739,13 @@ class CacheConfig:
             vLLM execution.
         swap_space: Size of the CPU swap space per GPU (in GiB).
         cache_dtype: Data type for kv cache storage.
+        is_attention_free: Whether the model is attention-free.
         num_gpu_blocks_override: Number of GPU blocks to use. This overrides the
             profiled num_gpu_blocks if specified. Does nothing if None.
+        sliding_window: Sliding window size for the KV cache. Can not work with
+            prefix caching enabled.
+        enable_prefix_caching: Whether to enable prefix caching.
+        cpu_offload_gb: Size of the CPU offload buffer in GB.
```
Comment (DarkLight1337) [inline vllm/config.py:748]: Should be GiB, see the docstring in `LLM` class.

**Review by DarkLight1337 — COMMENTED** [review]
Thanks for updating this, nevertheless I think that having to copy the documentation in several places is technical debt. @youkaichao what do you think?

**Comment by youkaichao** [comment]
We should make it a `dataclass` , and place the comment around the field.

this is just historical tech debt.

**Comment by wangxiyuan** [comment]
> We should make it a `dataclass` , and place the comment around the field.
> 
> this is just historical tech debt.

Agree, Mengqing Cao proposed  a RFC about config refactor, How about do this together? https://github.com/vllm-project/vllm/issues/10666

**Review by DarkLight1337 — APPROVED** [review]
LGTM now

**Pull request merged**